api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -379,6 +379,9 @@ type ManagedControlPlaneVirtualNetwork struct {
 	// +optional
 	ResourceGroup string `json:"resourceGroup,omitempty"`
 
+	// Name is the name of the virtual network.
+	Name string `json:"name"`
```

Comment:
If this ultimately overlaps with what's defined in the CRD already, maybe we don't want to specify the name here?